### DISCUSSION TITLE
Insulin usage in statistics within xDrip

### DISCUSSION THREAD

**Discussion by hupfpfupf** [NightscoutFoundation/xDrip#2390] [Ideas]
Hi There
i use the novopen 6 and am very happy with the synchronisation via NFC to xdrip.
I would however also be interested in having a statistic about the regular insulin usage (how many units of each insulin are used on average daily over the past 7 days, 30 days and 90 days). 
i have found the statistics tab in xdrip, however this only shows me the average glucose readings from my libre, nothing about the insulin. 
would this be feasible to implement? as all the data is synchronized in the app, i assume this should be rather straight forward?
or is it already available and i'm just not looking at the right spot?
Thanks for any support!

**Comment by Navid200** [comment]
I don't believe such a feature exists.
Regardless, I am curious why this would be useful.

The reason insulin dosage is changed is to minimize blood glucose variation, and to adjust the average blood glucose.
The variation of insulin is not an independent parameter with no impact on blood glucose.  
How would having your insulin dosage statistics help you exactly?

Let me ask my question using an example.
Let's say we have two individuals who both use insulin to control their blood glucose.
Looking at their blood glucose statistics, let's say we see identical perfect control, both with very small variation, both with perfect average.

Looking at their insulin intake statistics, we see that one takes more insulin than the other.  Does that realy matter?
All that means is that one is more sensitive to insulin than the other.
At the end, the only thing that matters is how well their blood glucose is controlled, isn't it?

**Reply by hupfpfupf** [reply]
hi. it is not for cross-people-comparison. it would be a longitudinal analysis of my own insuline usage. I am Type 1 diabetic, but many of my family memebers are type 2 diabetics. therefore, i am somewhat concerned about my insulin-sensitivity (i don't want both diabetes types :)). I eat quite unregularly, i sometimes snack or cannot have proper lunch due to work-hours and therefore don't have my fixed schemes. This works fine and my hba1c is good (<6.5), since i'm using the libre and can see my blood sugar always on my smartwatch in realtime.
My insulin-sensitivity varies quite a lot, depending on my physical activity. in weeks where i need to work a lot (at the desk) and barely get any exercise, i need much more insulin (where i usually correct 1-2 hours after lunch). a proper statistic could provide me with the information, depending on the week, how much i would anyways need and maybe adjust the basis-insuline instead of correcting with bolus-insulin. it would also allow me to early detect a decline in insuline sensitivity (which would be a clear sign to move more and maybe even to change jobs).
i know, i could do manual statistics. however since the feature is available to sync it to xdrip, i thought it would be neat to have it in there.
I also talked to my doctor about it, as he usually asks about average daily insuline intake, and i often provide him with rough estimates. He would be quite interested to have this porperly assessed.
Hope thats helping for understanding my thoughts?

**Comment by jamorham** [comment]
You can use the upload feature for Tidepool or Nightscout and get the statistics there but obviously its not an in-app solution. Its something that could be added but not a task that anyone is looking at at the moment as far as I know.

**Reply by hupfpfupf** [reply]
thank you! i will try out nightscout. i was not aware that the feature exists in there. 
and i still hope that someone will pick up the idea and implement it within xdrip, given the implementation would be quite easy :)
Best regards.

**Reply by Navid200** [reply]
I will add this if I can after we get through this:
https://github.com/NightscoutFoundation/xDrip/pull/3707

**Comment by Chromatticism** [comment]
I would just like to second that average daily insulin use would be a helpful statistic for several reasons:

- people who are in the honeymoon phase still produce some endogenous insulin and require less daily insulin, so tracking daily insulin use will help them anticipate when they are leaving the honeymoon phase and would require more intense management of their diabetes, something they could discuss with their healthcare team
- I am currently trying to lose weight and I have noticed a correlation between how much insulin I use and my weight so my current goal is to maintain my time in target while reducing my daily insulin use
- illness or infection will cause a decrease in insulin sensitivity therefore tracking daily insulin use could provide forewarning of an infection, or show when it is clearing up 
- I think the above poster also makes a very valid point that many type 1 diabetics develop increasing insulin resistance (essentially developing type 2 as well) which one could also monitor via average daily insulin use. Developing insulin resistance and requiring higher doses can make managing type 1 diabetes much more difficult.

For similar reasons daily carbohydrate consumption would also be helpful to track.

**Comment by Benjalic** [comment]
Again I would just like to third that average daily insulin use would be a helpful statistic.
Or some other basic stats from the nvopen read data would be usefull in my opinion.

**Comment by halarts** [comment]
Insulin pumps have limited insulin reservoirs that can vary from 200 (Omnipod) to 300 (Tandem). While the Tandem can be reloaded, the Omnipod cannot (you'd have to insert a new pod early). That might be an important factor in deciding which pump to choose.

And it seems odd that I can't just export to a CSV all of the data I have entered -- I can do that from the DEXCOM, but I have been entering it all directly into xdrip so that doesn't show up on my DEXCOM history.

**Comment by Teute0815** [comment]
There's a whole bunch of treatments, analysis, calculations, formulas and actions, that are based on TDD (Total Daily Dose). It's a central concept of Diabetes Management.

I do get my TDD via Nightscout Reporter but it would be very helpful to find it in xDrip Stats. 
I've been definitely missing it there, every time I was looking for it (quickly).

>How would having your insulin dosage statistics help you exactly?

calculating/reviewing carb ratios, correction factors, basal rates, bolus/basal ratio, insulin sensitivity - treating ketoacidosis - setting up AID insulin pumps, etc.
For more insights on TDD - google and ChatGPT will have some more answers.
